src/Common/src/System/Net/HttpKnownHeaderNames.TryGetHeaderName.cs

```diff
@@ -155,6 +155,7 @@ public static unsafe bool TryGetHeaderName(IntPtr buffer, int length, out string
                         case 'K': potentialHeader = KeepAlive; goto TryMatch;  // [K]eep-Alive
                         case 'S': potentialHeader = SetCookie; goto TryMatch;  // [S]et-Cookie
                         case 'U': potentialHeader = UserAgent; goto TryMatch;  // [U]ser-Agent
+                        case 'R': potentialHeader = RequestId; goto TryMatch;  // [R]equest-Id
```

Comment:
My primary concern with leaving it in the code is that it's taking up the potential place of another header we might want to add and would conflict with it.  Go ahead and leave it for now, but please add a comment next to it indicating that it's unexpected to be needed for responses and could be replaced by another case in the future if necessary.